components/core/src/clp_s/JsonParser.cpp

```diff
@@ -39,6 +40,199 @@ JsonParser::JsonParser(JsonParserOption const& option)
     m_archive_writer->open(archive_writer_option);
 }
 
+void JsonParser::parse_obj_in_array(ondemand::object line, int32_t parent_node_id) {
+    ondemand::object_iterator it = line.begin();
+    if (it == line.end()) {
+        m_current_schema.insert_unordered(parent_node_id);
+        return;
+    }
+
+    std::stack<ondemand::object> object_stack;
+    std::stack<ondemand::object_iterator> object_it_stack;
+
+    std::stack<int32_t> node_id_stack;
+    node_id_stack.push(parent_node_id);
+    object_stack.push(std::move(line));
+    object_it_stack.push(std::move(it));
+
+    size_t object_start = m_current_schema.start_unordered_object(NodeType::Object);
```

Comment:
Yes, this is necessary to act as delimeters for the nested array case.

Consider the array `[[0],[0]]` vs the array `[[0,0]]`, if we don't start_unordered_object for each level of nesting in the array these two cases become indistinguishable (because the schema would just be <array, 2><mst node id X><mst node id X>). We don't have this issue with objects because uniqueness of keys means they won't repeat the same mst node id.